### DISCUSSION TITLE
Holes in mhm and mrm outputs

### DISCUSSION THREAD

**Discussion by ViniciusBMoreira** [mhm-ufz/mHM#21] [Q&A] [answered]
Hello!

With your help, I was able to run the mhm for my own basin without errors/warnings. Thank you for that.

But my outputs (mhm and mrm fluxes) have pixels without data.
I tried to change parameters and redid some inputs, but without success.

Do you have any tips on what might solve this?

[mhm_outputs.nml.txt](https://github.com/mhm-ufz/mhm/files/6667028/mhm_outputs.nml.txt)
[mhm.nml.txt](https://github.com/mhm-ufz/mhm/files/6667029/mhm.nml.txt)
[mhm_parameter.nml.txt](https://github.com/mhm-ufz/mhm/files/6667031/mhm_parameter.nml.txt)
[mrm_outputs.nml.txt](https://github.com/mhm-ufz/mhm/files/6667032/mrm_outputs.nml.txt)

![image](https://user-images.githubusercontent.com/61054098/122322800-70886700-cefc-11eb-8983-dc7ef30405c6.png)

pet.nc, pre.nc and tavg.nc seem to have data consistent with reality and without holes

![image](https://user-images.githubusercontent.com/61054098/122323731-ef31d400-cefd-11eb-9355-543d094cb21a.png)

**Comment by pallavhydro** [comment]
Hi Bryan

Good to know you are making progress. Thanks for sharing the screenshots. My comments -

1. The mhm_fluxes_states figure (streamflow) definitely looks out of place. The chosen color scale, however, is not that helpful here. Based on this color scheme, both low values and nodata (holes) could be white. Could you please use a color scheme where white is not included and repost the screenshot?

2. Also please post such screenshot for facc.asc that you are using. The funny streamflow makes me doubt the L0 flow accumulation being used.

Cheers
Pallav

**Reply by ViniciusBMoreira** [reply]
Hey @pallavhydro. Sure!

1 -  
[mRM_Fluxes_States.nc.txt](https://github.com/mhm-ufz/mhm/files/6669961/mRM_Fluxes_States.nc.txt)


![image](https://user-images.githubusercontent.com/61054098/122392330-27f79a80-cf4a-11eb-9527-b4d86c73446f.png)

2 - 
[facc.asc.txt](https://github.com/mhm-ufz/mhm/files/6669966/facc.asc.txt)


![image](https://user-images.githubusercontent.com/61054098/122391860-ba4b6e80-cf49-11eb-8572-99b1e5a44975.png)

3- 
[fdir.asc.txt](https://github.com/mhm-ufz/mhm/files/6669971/fdir.asc.txt)

![image](https://user-images.githubusercontent.com/61054098/122393089-e61b2400-cf4a-11eb-9df2-dd8ff33094cc.png)

Curious:

![image](https://user-images.githubusercontent.com/61054098/122489209-8017b680-cfb5-11eb-8e3b-ad92bac57b5c.png)

I made a run with l0=l1=l11=500m, and this was the results:

![image](https://user-images.githubusercontent.com/61054098/122499442-510a4080-cfc7-11eb-9798-de61f3d60092.png)

And yes, the no data values are in the main channel:

![image](https://user-images.githubusercontent.com/61054098/122499481-64b5a700-cfc7-11eb-8bba-24032e75caf5.png)

**Comment by rohinikumar82** [comment]
Hi @BryanSouzaM,
Sorry for coming late on this issue. It is quite strange error not yet encountered...  

Could you please plot the DEM (the base file) and also "facc.asc" with some detail colouring.  Also could you save the mRM restart file and see what FACC and basin mask you are getting for your configuration. 

Best,
Rohini

**Reply by ViniciusBMoreira** [reply]
Hey @rohinikumar82  !


First of all, thank you **so much** for your support. I am using the model within my university for my final paper using a basin that is currently experiencing a [historic drought crisis.](https://www.em.com.br/app/noticia/economia/2021/05/29/internas_economia,1271519/estiagem-afeta-reservatorios-e-cemig-projeta-seca-historica.shtml)

If you prefer, I can upload my entire folder to Github!

Here is the information you asked for:

### dem.asc (base file)

![image](https://user-images.githubusercontent.com/61054098/122618084-e3602200-d063-11eb-8fb3-a66516ef4593.png)

### facc.asc (base file)

![image](https://user-images.githubusercontent.com/61054098/122618186-1dc9bf00-d064-11eb-9ff7-72d25d608fd7.png)

### L0_fAcc (mRM restart file) (yes, it's "upside down")

![image](https://user-images.githubusercontent.com/61054098/122618344-7600c100-d064-11eb-8c62-6c063df64632.png)

### L0_domain_mask

![image](https://user-images.githubusercontent.com/61054098/122618523-cc6dff80-d064-11eb-8a7f-1c983df2e606.png)


### L11_fAcc
![image](https://user-images.githubusercontent.com/61054098/122618653-0b9c5080-d065-11eb-85c2-69713605aae6.png)

### L11_domain_mask

![image](https://user-images.githubusercontent.com/61054098/122618689-21117a80-d065-11eb-8c00-803fc0cf62af.png)

### Correct location of basin:

![image](https://user-images.githubusercontent.com/61054098/122618718-38506800-d065-11eb-9e11-93cabd69f002.png)

There is holes in this files of restart ones:

L11_qTR
L11_Qmod
L11_qTIN

empty files:

L11_nLinkFracFPimp
L11_L1_ld


Thanks a lot!
Bryan

**Comment by rohinikumar82** [comment]
Hi @BryanSouzaM,
thanks for these pics... they all points to me that everything well - mHM-mRM doing its job as expected - as you can see L11_fAcc is correctly showing the upscaled river network. Don't worry about the upside down flipping of the mHM/mRM outputs - all is well here just that this is the way, mHM is currently writing its outputs.

Now lets hunt back your issue with outputs of "mRM_fluxes_states" -  I see (from your setting in mhm output nml file) that you are writing the routed streamflow - which comes as an output written in a netcdf file (with three dimensions variable [x, y, time]). Here my question is which time-snap (day) are your printing (and showing) here as the spatial plot? Hope there is no issue with dimension while plotting?    

Can u also save mHM_fluxes_sates - there make the total_runoff, ET --> .TRUE.; and plot the first tilmestep total_runoff plot

thanks. 

Best,
Rohini

**Reply by ViniciusBMoreira** [reply]
Hey @rohinikumar82 

I was plotting the first day of the simulation and I can plot any of the period.

**Important!:**  As I change the simulation day, the number of cells with no data changes, for mRM and mHM.


Total runoff for day1:

![image](https://user-images.githubusercontent.com/61054098/122620984-b19e8980-d06a-11eb-9657-f3eb9f83e83e.png)


![image](https://user-images.githubusercontent.com/61054098/122620741-17d6dc80-d06a-11eb-8443-d46b53332f22.png)

**Comment by rohinikumar82** [comment, marked as answer]
Could you please run the mhm with changes in "mhm.nml" as timestep = 1 !! instead of = 24

and in the "mrm_outputs.nml" file for timeStep_model_outputs_mrm = 0

**Reply by ViniciusBMoreira** [reply]
Sure, a sec.
Bryan

**Reply by ViniciusBMoreira** [reply]
Hey @rohinikumar82 !

I just got one period of time, but again, with holes.

I can upload my folder to github, with you prefer.

![image](https://user-images.githubusercontent.com/61054098/122621887-4bffcc80-d06d-11eb-9b94-5d8f45a6afca.png)

Cheers,
Bryan

**Reply by rohinikumar82** [reply]
thanks can u share the datasets - I will give a try to  hunt this bug !

**Reply by rohinikumar82** [reply]
btw: which mHM version you are using?

**Reply by ViniciusBMoreira** [reply]
@rohinikumar82 i'm using 5.11.2-dev0, acording to version.txt

I will upload my folder on github

**Reply by rohinikumar82** [reply]
can u also try to run your basin with v. 5.11.1, which is so-far the released version.

**Reply by ViniciusBMoreira** [reply]
Sure!

**Reply by ViniciusBMoreira** [reply]
@rohinikumar82 , v.5.11.1 give the same results

**Reply by rohinikumar82** [reply]
hmmm... lets hunt this issue... part I am looking at your nml file.. you may consider changing some set-up (with soil layer etc).. but I do not think this should be an issue to the current problem at hand... 

btw: I could not see the total_runoff (mhm_flux_states) pic. earlier (does that also has holes?)

**Reply by ViniciusBMoreira** [reply]
@rohinikumar82 ,

here is the link to a folder on the google drive, if you prefer.

Thank you so much for the support so far.

Bryan.

**Reply by ViniciusBMoreira** [reply]
> hmmm... lets hunt this issue... part I am looking at your nml file.. you may consider changing some set-up (with soil layer etc).. but I do not think this should be an issue to the current problem at hand...
> 
> btw: I could not see the total_runoff (mhm_flux_states) pic. earlier (does that also has holes?)

Yes! There is holes

![image](https://user-images.githubusercontent.com/61054098/122623009-f2999c80-d070-11eb-9214-dbfe84e11eae.png)

**Reply by rohinikumar82** [reply]
got the files... looking at them...

**Reply by rohinikumar82** [reply]
can u tell me created soil_datasets? there appears to be some sort of issues. Before we go into this one: 

to debug the current problem in hand
can u please change in file "soil_classdefinition_iFlag_soilDB_1.txt" 

in sand/clay % coloums [2 and 3] where it "0" to "1" 
and for Bd "0" to "1.33" 

run the code and let me know the outcome

btw: the holes are not really the missing values but a very big number

**Reply by ViniciusBMoreira** [reply]
sure! a sec @rohinikumar82

**Reply by rohinikumar82** [reply]
there are also some other configuration settings that deserve some attentions!
Lets solve the issue at hand first!

**Reply by rohinikumar82** [reply]
I think soil is an issue... your interception flux (in mHM_fluxes_states file) is right in extent (and no holes there)! So everything after soil module would be corrupted!

**Reply by ViniciusBMoreira** [reply]
@rohinikumar82 

I think its worked !!!!

There is some bad clips, because i changed the shape of the basin trying to correct this!

![image](https://user-images.githubusercontent.com/61054098/122623788-dcd9a680-d073-11eb-9b91-906ccbb214ae.png)

**Reply by ViniciusBMoreira** [reply]
Sure @rohinikumar82 ! I can create one. No problem.
I can also invite you on Microsoft Teams

**Reply by rohinikumar82** [reply]
fine!

**Reply by rohinikumar82** [reply]
> @rohinikumar82
> 
> I think its worked !!!!
> 
> There is some bad clips, because i changed the shape of the basin trying to correct this!
> 
> ![image](https://user-images.githubusercontent.com/61054098/122623788-dcd9a680-d073-11eb-9b91-906ccbb214ae.png)

happy it worked!

**Reply by pallavhydro** [reply]
> @rohinikumar82 
> 
> 
> 
> I think its worked !!!!
> 
> 
> 
> There is some bad clips, because i changed the shape of the basin trying to correct this!
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/61054098/122623788-dcd9a680-d073-11eb-9b91-906ccbb214ae.png)
> 
> 

Hi Bryan.

Great that this worked. And your determination to work with mHM is impressive to say the least 😊 

Could you perhaps add a clear note on what was the solution (**bad clips**?), for future reference and other users? Thanks in advance.

Pallav

**Reply by ViniciusBMoreira** [reply]
@pallavhydro @rohinikumar82 

I couldn't do it without your help! finally I can have fun calibrating the model!!

### For new users:

You should not leave values ​​of 0 in soil tables (sand, clay and bd), this leads to infinite values ​​in mHM. Instead, use at least 1% values ​​for clay and sand and greater than 1 for bd (for bd, the minimum value I'm using is 1.17)

Thanks a lot!
Bryan
